Week3/homework/readme.md

```diff
@@ -0,0 +1,115 @@
+- Install all the dependencies: open `homework\package.json` in the terminal and run the command `npm install`.
+- Run the file `homework\server.js`.
+
+Now in the terminal  you have to see somthing like this:
+ `The server is listening to http://localhost:3000`
+`connected as id 515` (Sure with different id number)
+If you are seeing that then everything is good and you can go ahead.
+# Testing the app
+Run the PostMan program on your computer, so you can test the app by sending HTTP requests to the endpoints/URLs
+#### ## 1- Create a new ToDo list:
+**Insert the folowing in PostMan then send the request:**
+- HTTP method: POST.
+- Endpoint: `http://localhost:3000/lists`
+- Request body: raw.
+- Request body conent type: `JSON`.
+- Request body: `{"listName":"test 1","userIdReference":1}`
+- Send the request.
+- You will get response likie this:
+ `{ "message": "Trying to add new list to the database.",
+    "info": "Added successfully, The list name is: test 1, for the user with ID: 1"}`
+- In the database you can see that one list is added, and it has ID 10.
+#### ## 2- Delete a ToDo list:
+**Insert the folowing in PostMan then send the request:**
+- HTTP method: DELETE.
+- Endpoint syntax:`http://localhost:3000/lists/{id}`
+- Endpoint: `http://localhost:3000/lists/10`. The number 10 references to the list ID.
+- Send the request.
+- You will get response likie this:
+ `{ "message": "Trying to delete list with ID: 10.",
+    "info": "The list is deleted successfully, with all it's related tasks "}`
+- In the database you can see that the list is deleted.
+#### ## 3- Add a reminder for a list:
+The reminder is a new date after few days from now. in the body request you will send the number of days.
+**Insert the folowing in PostMan then send the request:**
+- HTTP method: PATCH.
+- Endpoint syntax:`http://localhost:3000/lists/{id}`
+- Endpoint: `http://localhost:3000/lists/1` The number 1 references to the list ID.
+- Request body: raw.
+- Request body conent type: `JSON`.
+- Request body: `{"listReminder":10}`
+- Send the request.
+- You will get response likie this:
+ `{"message": "Trying to add reminder to the list with ID 1",
+    "reminderSetting": "Remind after 10 days, then it will be Tue Oct 08 2019 06:41:22 GMT+0200 (GMT+02:00)",
+    "info": "The reminder is added successfully"}`
+- In the database you can see that the list with id 1 has a reminder.
+#### ## 4- Insert task(s) in ToDo list:
+**Insert the folowing in PostMan then send the request:**
+- HTTP method: POST.
+- Endpoint: `http://localhost:3000/tasks`
+- Request body: raw.
+- Request body conent type: `JSON`.
+- Request body:
+`[ {"taskDescription": "new task 1", "listIdReference": 1  },{"taskDescription": "new task 2", "listIdReference": 1  },{"taskDescription": "new task 3", "listIdReference": 1  }]`
+- Send the request.
+- You will get response likie this:
+ `{    "message": "Adding task or tasks done successfully",
```

Comment:
Great error message. We could improve it even more by adding the new ID of the item created. Something like for instance: "Adding task or task done successfully with id=20"

Reply:
yes, I though about the same thing.. The result object is very helpful for that target.
I couldn't apply all my thoughts, I have ben already late to submit the homework.